index.js

```diff
@@ -21,14 +21,17 @@ var toArray = Array.from || (function (slice) {
 
 // ===================================================================
 
-function eventToPromise (emitter, event, opts) {
-  var ignoreErrors = opts && opts.ignoreErrors
+function eventToPromise (emitter, event, _opts) {
+  var opts = _opts || {}
+  var ignoreErrors = opts.ignoreErrors
+  var errorEvent = opts.error || 'error'
 
   return new Bluebird(function (resolve, reject) {
     var addListener =
+      emitter.once ||
+      emitter.on ||
```

Comment:
TBH, I'm not sure why these checks exist. EventEmitter is an interface i.e. an event emitter is something with the same API as EventEmitter (with possible additions).

Did you have a particular incompatible implementation in mind when including these checks?